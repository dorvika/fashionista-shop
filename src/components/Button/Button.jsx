import React from "react";

const Button = ({ className, onClick, children, type }) => {
  return (
    <button className={className} onClick={onClick} type={type}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  className: null,
  type: "button",
};

export default Button;
