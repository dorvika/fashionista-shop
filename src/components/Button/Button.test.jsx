import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

describe("<Button /> component", () => {
  it("should render button tag", () => {
    render(<Button>Confirm</Button>);
    screen.getByRole("button");
  });
  it("should render button with text", () => {
    render(<Button>Buy</Button>);
    screen.getByText(/buy/i);
  });
  it("should pass onClick if button is clicked", () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click Me</Button>);
    fireEvent.click(screen.getByText(/click me/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  it("should match snapshot", () => {
    const { container } = render(<Button>Confirm</Button>);
    expect(container).toMatchSnapshot();
  });
});
