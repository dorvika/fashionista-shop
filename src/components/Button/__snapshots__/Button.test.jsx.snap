// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component should match snapshot 1`] = `
<div>
  <button
    type="button"
  >
    Confirm
  </button>
</div>
`;

exports[`<Button /> component should render 1`] = `
<div>
  <button
    type="button"
  >
    Confirm
  </button>
</div>
`;
